### DISCUSSION TITLE
`vite dev` vs. customized`server/index.ts`, which to use?

### DISCUSSION THREAD

**Discussion by upupming** [vikejs/vike#1351] [Discussion] [answered]
Thanks for your great framework, I like Vike very much!

After diving into the source code of Vike, I have a question. In the `boilerplates` examples, all the boilerplates use `node --loader ts-node/esm ./server/index.ts` to run the dev server, but in the `examples`, most of the use `vite` directly (eg. [react-full-v1](https://github.com/vikejs/vike/blob/27d13be4ee4de71666f2e6f889e8c7e26648e133/examples/react-full-v1/package.json#L3)) for the dev server, and still some are using `server/index.ts` (eg. [path-aliases](https://github.com/vikejs/vike/blob/main/examples/path-aliases/package.json#L10)).

1. So I am curious about which one should I use. `vite dev` or customized `server/index.ts`, As `boilerplates` are used by `create-vike`, does it mean that Vike recommends users use the customized `server.ts` to run their own dev server?
2. If we use `server/index.ts`, the file is transpiled by ts-node but not by Vite, thus we need a manual restart of vite when modifying it. If we introduce our `vike` CLI as https://github.com/vikejs/vike/issues/1340, should we support loading this `server/index.ts` file by Vite?

**Comment by brillout** [comment, marked as answer]
I usually boils down to that: either you pre-render or your use a server, see explanation at https://vike.dev/pre-rendering.

> should we support loading this server/index.ts file by Vite?

We're actually currently working on it: https://github.com/vikejs/vike/pull/1342.
